src/data_source/storage/fs.rs

```diff
@@ -291,12 +323,9 @@ where
         Ok(())
     }
 
-    fn revert(mut self) -> impl Future + Send {
-        async move {
-            self.inner.leaf_storage.revert_version().unwrap();
-            self.inner.block_storage.revert_version().unwrap();
-            self.inner.vid_storage.revert_version().unwrap();
-        }
+    fn revert(self) -> impl Future + Send {
+        // The revert is handled when `self` is dropped.
+        async move {}
```

Comment:
I think we should include the revert here as well. If this function is called anywhere, it's expected to trigger a revert

Reply:
Sorry, comment is not clear. `self` is dropped in this function, so without doing anything here, it does trigger a revert (if we reverted explicitly it would revert twice)